core/ErrorQueue.h

```diff
@@ -17,7 +17,7 @@ class ErrorQueue {
     std::vector<std::unique_ptr<ErrorQueueMessage>> drainAll();
     std::vector<std::unique_ptr<ErrorQueueMessage>> drainFlushed();
     void collectForFile(core::FileRef whatFile, std::vector<std::unique_ptr<core::ErrorQueueMessage>> &out);
-    ErrorFlusher errorFlusher;
+    std::unique_ptr<ErrorFlusher> errorFlusher;
```

Comment:
Let's actually make this a `shared_ptr`; we don't need to ensure unique ownership here, and it'll make downcasting easier if we ever need to.